src/testFixtures/java/org/opensearch/knn/TestUtils.java

```diff
@@ -327,10 +332,46 @@ private float[][] readQueries(String path) throws IOException {
             return queryArray;
         }
 
+        private void initBinaryData() {
+            // Find medium value
+            List<Float> flattenedVectors = new ArrayList<>(indexData.vectors.length * indexData.vectors[0].length);
+            for (int i = 0; i < indexData.vectors.length; i++) {
+                for (int j = 0; j < indexData.vectors[i].length; j++) {
+                    flattenedVectors.add(indexData.vectors[i][j]);
+                }
+            }
+            Collections.sort(flattenedVectors);
+            Float median = flattenedVectors.get(flattenedVectors.size() / 2);
+
+            // Quantize + Packing for index data
+            indexBinaryData = new byte[indexData.vectors.length][(indexData.vectors[0].length + 7) / 8];
```

Comment:
Can you add a comment explaining how you are quantizing and generating the data

Reply:
Quantization is obvious from the code I guess? `indexData.vectors[i][j] >= median ? 1 : 0`
Packing is also nothing special. Just packing 8 binary values into a single byte value.

Reply:
Let me add a comment.